pypiserver/core.py

```diff
@@ -256,6 +256,22 @@ def _listdir(root):
                               relfn=fn[len(root) + 1:])
 
 
+def read_lines(filename):
+    """
+    Read the contents of `filename`, stripping empty lines and '#'-comments.
+    Return a list of strings, containing the lines of the file.
+    """
+    lines = []
+
+    try:
+        with open(filename) as f:
+            lines = [line.strip() for line in f.readlines() if line.strip() and not line.strip().startswith('#')]
+    except Exception:
+        pass
```

Comment:
We may want to do a `log.warning` here or something, just so that there's some indication if the file failed to parse. I wonder if we might even want to exit on in the case of failure, if this is the sort of thing people are enabling for the sake of security

Reply:
That's probably a design / usability decision; personally I think there's too much space in `open()` to raise an exception and crash the whole pypiserver instance. Wrong file mode, privileges, path - just to name the most common ones. But you're absolutely right about the logging. I'll commit the change. 

If you or one of the maintainers decide that we don't want to start the server unless we read the blacklist file successfully, it should take just a `raise` statement to implement. I'm also more inclined to the security-before-availability approach, but don't want to take the burden of deciding this :slightly_smiling_face:

Reply:
I totally understand!

I thought this through a bit today, and I think we should bomb if someone specifies a non-existent file or a file can't be read. That way, we can't get into a situation where someone thinks they're successfully preventing updates, but they aren't actually. If you want to go ahead and merge this, I can add the `raise` afterwards. Or feel free to add it if you're feeling like it.

Thanks again for the patch! We'll probably get a new version out here sometime in the next week.

Reply:
Sure, I added the `raise` with a bit of an explanation. Looking forward for the update! :slightly_smiling_face: